File: Inventory.Api/Permissions/InventoryPermissionKeys.cs
namespace Inventory.Api.Permissions;

public static class InventoryPermissionKeys
{
    public const string DashboardView = "Inventory.Dashboard.View";

    public const string ReceiptsView = "Inventory.Receipts.View";
    public const string ReceiptsCreate = "Inventory.Receipts.Create";
    public const string ReceiptsEdit = "Inventory.Receipts.Edit";
    public const string ReceiptsReceive = "Inventory.Receipts.Receive";
    public const string ReceiptsApprove = "Inventory.Receipts.Approve";
    public const string ReceiptsReject = "Inventory.Receipts.Reject";
    public const string ReceiptsCancel = "Inventory.Receipts.Cancel";

    public const string ReceiptRejectionsView = "Inventory.ReceiptRejections.View";
    public const string ReceiptRejectionsResolve = "Inventory.ReceiptRejections.Resolve";

    public const string IssuesView = "Inventory.Issues.View";
    public const string IssuesCreate = "Inventory.Issues.Create";
    public const string IssuesEdit = "Inventory.Issues.Edit";
    public const string IssuesAllocate = "Inventory.Issues.Allocate";
    public const string IssuesPost = "Inventory.Issues.Post";
    public const string IssuesCancel = "Inventory.Issues.Cancel";

    public const string TransfersView = "Inventory.Transfers.View";
    public const string TransfersCreate = "Inventory.Transfers.Create";
    public const string TransfersEdit = "Inventory.Transfers.Edit";
    public const string TransfersAllocate = "Inventory.Transfers.Allocate";
    public const string TransfersShip = "Inventory.Transfers.Ship";
    public const string TransfersReceive = "Inventory.Transfers.Receive";
    public const string TransfersComplete = "Inventory.Transfers.Complete";
    public const string TransfersCancel = "Inventory.Transfers.Cancel";

    public const string AdjustmentsView = "Inventory.Adjustments.View";
    public const string AdjustmentsCreate = "Inventory.Adjustments.Create";
    public const string AdjustmentsEdit = "Inventory.Adjustments.Edit";
    public const string AdjustmentsPost = "Inventory.Adjustments.Post";
    public const string AdjustmentsCancel = "Inventory.Adjustments.Cancel";

    public const string CostsView = "Inventory.Costs.View";
    public const string CostsEdit = "Inventory.Costs.Edit";

    public const string ShelvesView = "Inventory.Shelves.View";
    public const string ShelvesManage = "Inventory.Shelves.Manage";
    public const string OperationsMoveStock = "Inventory.Operations.MoveStock";

    public const string WarehouseView = "Inventory.Warehouse.View";
    public const string WarehouseCreate = "Inventory.Warehouse.Create";
    public const string WarehouseEdit = "Inventory.Warehouse.Edit";
    public const string WarehouseActivate = "Inventory.Warehouse.Activate";

    public const string StockLedgerView = "Inventory.StockLedger.View";
    public const string StockItemsView = "Inventory.StockItems.View";
    public const string StockItemsSerialsView = "Inventory.StockItems.Serials.View";
    public const string StockItemsUnassignedView = "Inventory.StockItems.Unassigned.View";
    public const string StockItemsProductsView = "Inventory.StockItems.Products.View";

    public const string ReservationsView = "Inventory.Reservations.View";
    public const string ReservationsManage = "Inventory.Reservations.Manage";

    public const string CatalogProxy = "Inventory.Catalog.Proxy";
    public const string ReportsView = "Inventory.Reports.View";
    public const string AuditView = "Inventory.Audit.View";
}

File: InventoryAdmin.Web/src/app/inventoryPermissionKeys.ts
export const InventoryPermissionKeys = {
  DashboardView: 'Inventory.Dashboard.View',

  ReceiptsView: 'Inventory.Receipts.View',
  ReceiptsCreate: 'Inventory.Receipts.Create',
  ReceiptsEdit: 'Inventory.Receipts.Edit',
  ReceiptsReceive: 'Inventory.Receipts.Receive',
  ReceiptsApprove: 'Inventory.Receipts.Approve',
  ReceiptsReject: 'Inventory.Receipts.Reject',
  ReceiptsCancel: 'Inventory.Receipts.Cancel',

  ReceiptRejectionsView: 'Inventory.ReceiptRejections.View',
  ReceiptRejectionsResolve: 'Inventory.ReceiptRejections.Resolve',

  IssuesView: 'Inventory.Issues.View',
  IssuesCreate: 'Inventory.Issues.Create',
  IssuesEdit: 'Inventory.Issues.Edit',
  IssuesAllocate: 'Inventory.Issues.Allocate',
  IssuesPost: 'Inventory.Issues.Post',
  IssuesCancel: 'Inventory.Issues.Cancel',

  TransfersView: 'Inventory.Transfers.View',
  TransfersCreate: 'Inventory.Transfers.Create',
  TransfersEdit: 'Inventory.Transfers.Edit',
  TransfersAllocate: 'Inventory.Transfers.Allocate',
  TransfersShip: 'Inventory.Transfers.Ship',
  TransfersReceive: 'Inventory.Transfers.Receive',
  TransfersComplete: 'Inventory.Transfers.Complete',
  TransfersCancel: 'Inventory.Transfers.Cancel',

  AdjustmentsView: 'Inventory.Adjustments.View',
  AdjustmentsCreate: 'Inventory.Adjustments.Create',
  AdjustmentsEdit: 'Inventory.Adjustments.Edit',
  AdjustmentsPost: 'Inventory.Adjustments.Post',
  AdjustmentsCancel: 'Inventory.Adjustments.Cancel',

  CostsView: 'Inventory.Costs.View',
  CostsEdit: 'Inventory.Costs.Edit',

  ShelvesView: 'Inventory.Shelves.View',
  ShelvesManage: 'Inventory.Shelves.Manage',
  OperationsMoveStock: 'Inventory.Operations.MoveStock',

  WarehouseView: 'Inventory.Warehouse.View',
  WarehouseCreate: 'Inventory.Warehouse.Create',
  WarehouseEdit: 'Inventory.Warehouse.Edit',
  WarehouseActivate: 'Inventory.Warehouse.Activate',

  StockLedgerView: 'Inventory.StockLedger.View',
  StockItemsView: 'Inventory.StockItems.View',
  StockItemsSerialsView: 'Inventory.StockItems.Serials.View',
  StockItemsUnassignedView: 'Inventory.StockItems.Unassigned.View',
  StockItemsProductsView: 'Inventory.StockItems.Products.View',

  ReservationsView: 'Inventory.Reservations.View',
  ReservationsManage: 'Inventory.Reservations.Manage',

  CatalogProxy: 'Inventory.Catalog.Proxy',
  ReportsView: 'Inventory.Reports.View',
  AuditView: 'Inventory.Audit.View',
} as const

export type InventoryPermissionKey = typeof InventoryPermissionKeys[keyof typeof InventoryPermissionKeys]
